alpha-build-machinery/make/targets/openshift/deps-gomod.mk

```diff
@@ -0,0 +1,41 @@
+self_dir :=$(dir $(lastword $(MAKEFILE_LIST)))
+
+# We need to force localle so different envs sort files the same way for recursive traversals
+deps_diff :=LC_COLLATE=C diff --no-dereference -N
+
+# $1 - temporary directory
+define restore-deps
+	ln -s $(abspath ./) "$(1)"/current
+	cp -R -H ./ "$(1)"/updated
+	$(RM) -r "$(1)"/updated/vendor
+	cd "$(1)"/updated && $(GO) mod tidy && $(GO) mod vendor && $(GO) mod verify
```

Comment:
Shouldn't the order by vendor and then tidy?

Reply:
As it modifies the go.mod I see `go mod tidy` as first step. It also adds dependencies and prune them which should logically preceed commit them to vendor folder.
```
go mod tidy finds all the packages transitively imported by packages in your module. It adds new module requirements for packages not provided by any known module, and it removes requirements on modules that don't provide any imported packages. If a module provides packages that are only imported by projects that haven't migrated to modules yet, the module requirement will be marked with an // indirect comment. It is always good practice to run go mod tidy before committing a go.mod file to version control.
```
https://blog.golang.org/migrating-to-go-modules